src/pumipic_adjacency.tpp

```diff
@@ -544,5 +548,42 @@ namespace pumipic {
     Kokkos::Profiling::popRegion();
     return found;
   }
+
+  template <typename ParticleType, typename Segment3d>
+  struct NativeParticleHandlerAtElemBdry {
+    NativeParticleHandlerAtElemBdry () {}
+
+    void operator()(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
+                    o::Write<o::LO>& elem_ids, o::Write<o::LO>& inter_faces,
+                    o::Write<o::LO>& lastExit, o::Write<o::Real>& inter_points,
+                    o::Write<o::LO>& ptcl_done,
+                    bool requireIntersection,
```

Comment:
`requireIntersection` should not be an input to `operator()`, but member data isntead.